ios/browser/api/password/brave_password_api.h

```diff
@@ -0,0 +1,123 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_PASSWORD_BRAVE_PASSWORD_API_H_
+#define BRAVE_IOS_BROWSER_API_PASSWORD_BRAVE_PASSWORD_API_H_
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, PasswordFormScheme) {
+  PasswordFormSchemeTypeHtml = 0,
+  PasswordFormSchemeTypeBasic,
+  PasswordFormSchemeTypeDigest,
+  PasswordFormSchemeTypeOther,
+  PasswordFormSchemeMobile,
+  PasswordFormSchemeUsernameOnly
+};
+
+@protocol PasswordStoreObserver;
+@protocol PasswordStoreListener;
+
+@class IOSPasswordForm;
+
+NS_SWIFT_NAME(PasswordForm)
+OBJC_EXPORT
+@interface IOSPasswordForm : NSObject <NSCopying>
+
+@property(nonatomic, strong) NSURL* url;
+@property(nonatomic, strong) NSString* signOnRealm;
+@property(nonatomic, nullable, copy) NSDate* dateCreated;
+@property(nonatomic, nullable, copy) NSDate* dateLastUsed;
+@property(nonatomic, nullable, copy) NSDate* datePasswordChanged;
+@property(nonatomic, nullable, copy) NSString* usernameElement;
+@property(nonatomic, nullable, copy) NSString* usernameValue;
+@property(nonatomic, nullable, copy) NSString* passwordElement;
+@property(nonatomic, nullable, copy) NSString* passwordValue;
+@property(nonatomic) bool isBlockedByUser;
+@property(nonatomic) PasswordFormScheme scheme;
+
+/// Password Form Constructor used with BravePasswordAPI
+/// @param url - Primary data used by the PasswordManager to decide (in longest
+///        matching prefix fashion) whether or not a given PasswordForm
+///        result from the database is a good fit for a particular
+///        form on a page
+/// @param signOnRealm - The "Realm" for the sign-on. This is scheme, host,
+///        port for SCHEME_HTML
+///        The signon_realm is effectively the primary key used for retrieving
+///        data from the database, so it must not be empty
+/// @param dateCreated - The date when login was saved
+/// @param dateLastUsed - The date when the login was last used by the user
+/// to login to the site
+/// @param datePasswordChanged - Date when the password value was last changed
+/// @param usernameElement - The string represantation of the name of the
+/// username input element
+/// @param usernameValue - The string represantation of the username
+/// @param passwordElement - The string represantation of the name of the
+/// password input element
+/// @param passwordValue - The string represantation of the password
```

Comment:
minor: `representation` not `represantation`

Reply:
Fix at https://github.com/brave/brave-core/pull/12360/commits/00549d0c3e8bb7dc718f02714395a6a423d1a6ab